### DISCUSSION TITLE
Google OAuth/Next Auth won't create new users when login in with multiple accounts.

### DISCUSSION THREAD

**Discussion by kaijuh** [nextauthjs/next-auth#11626] [Help]
Hello,

Below are my system specs and packages

```

System:
    OS: Linux 4.4 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    CPU: (24) x64 12th Gen Intel(R) Core(TM) i9-12900K
    Memory: 47.67 GB / 63.75 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.12.2 - ~/.nvm/versions/node/v20.12.2/bin/node
    npm: 10.8.2 - ~/.nvm/versions/node/v20.12.2/bin/npm
    pnpm: 9.7.1 - ~/.local/share/pnpm/pnpm
  npmPackages:
    @auth/pg-adapter: ^1.4.2 => 1.4.2
    next: 15.0.0-canary.108 => 15.0.0-canary.108
    next-auth: 5.0.0-beta.20 => 5.0.0-beta.20
    react: 19.0.0-beta-4508873393-20240430 => 19.0.0-beta-4508873393-20240430
```
    
 I set up Google Auth and I am able to login and get user data. But the problem is that even though I am login in with different google accounts, I still only see one account created.

![image](https://github.com/user-attachments/assets/f0c7dae3-0678-4362-8061-70223ef28260)

However, when I check my 3rd party connections in all of my other google accounts, I do see an authorized connection made to the app.

![image](https://github.com/user-attachments/assets/68c24b22-2a68-4661-a8e3-894b624716ea)
![image](https://github.com/user-attachments/assets/b4d13169-0b74-49d9-80a0-15accd77d8d9)

For my storage I am using 
https://vercel.com/docs/storage/vercel-postgres
    
Also my verification token table in my postgres vercel db is empty so I am guessing that is probably where the issue lies.  
![image](https://github.com/user-attachments/assets/c0736d30-cdec-47f9-8fa8-205991eb9888)

I used the SQL query below to create my tables found in here (  https://authjs.dev/getting-started/adapters/pg)

```
CREATE TABLE verification_token
(
  identifier TEXT NOT NULL,
  expires TIMESTAMPTZ NOT NULL,
  token TEXT NOT NULL,
 
  PRIMARY KEY (identifier, token)
);
 
CREATE TABLE accounts
(
  id SERIAL,
  "userId" INTEGER NOT NULL,
  type VARCHAR(255) NOT NULL,
  provider VARCHAR(255) NOT NULL,
  "providerAccountId" VARCHAR(255) NOT NULL,
  refresh_token TEXT,
  access_token TEXT,
  expires_at BIGINT,
  id_token TEXT,
  scope TEXT,
  session_state TEXT,
  token_type TEXT,
 
  PRIMARY KEY (id)
);
 
CREATE TABLE sessions
(
  id SERIAL,
  "userId" INTEGER NOT NULL,
  expires TIMESTAMPTZ NOT NULL,
  "sessionToken" VARCHAR(255) NOT NULL,
 
  PRIMARY KEY (id)
);
 
CREATE TABLE users
(
  id SERIAL,
  name VARCHAR(255),
  email VARCHAR(255),
  "emailVerified" TIMESTAMPTZ,
  image TEXT,
 
  PRIMARY KEY (id)
);
```

The only empty table in my db is verification_token

**Comment by kaijuh** [comment]
**UPDATE**
I was **now** able to create new users by login with **different** google accounts which were **not signed in** when I initially logged in for the first time.

For example, 

if one of my other google accounts was signed out, and I chose to sign in with that **signed out account**, then the user was created with out issues.
![image](https://github.com/user-attachments/assets/d7627dc3-8445-4ca0-9c21-4bb6b14aaf76)
But the problem was with accounts that were **already** signed in.

This almost feels like a bug but it could likely be due to how the session is being handled.

**Reply by kaijuh** [reply]
UPDATE 2:

I was now able to create new users with the two original accounts that were linked to the first id.

In my accounts table I noticed that they were 3 separate accounts (1 , 2 ,3)  that all had the same userId. 
Those 3 accounts connected to the 3 different email addresses that were signed in when I first initially connected my app to the main email.

It looks like Next Auth created 3 separate rows in my accounts table using the same id of the users table. So the other 2 emails were always linking back to the first email.

What I did to solve the issue was to delete the 2 accounts that were linked back to the first userId.

After doing that, I was now able to create a new user and a new account.  I just dont see how creating 3 signed accounts with 1 email sign in could have happened. I found another thread which could mean this might be a bug.

https://github.com/nextauthjs/next-auth/discussions/3795

**Reply by damain** [reply]
@kaijuh there is a sign in callback that you can use. Maybe you can do any checks for user accounts there. But hopefully you won't need it if your issue is already sorted

https://next-auth.js.org/configuration/callbacks
